core/src/main/scala/kanaloa/reactive/dispatcher/queue/QueueProcessor.scala

```diff
@@ -21,6 +21,11 @@ class QueueProcessor(
   resultChecker:          ResultChecker
 ) extends Actor with ActorLogging with MessageScheduler {
 
+  val healthCheckSchedule = {
```

Comment:
Don't we kind of already have the infinite loop?  IE: if every HealthCheck fails, you will try to spawn more Routees?   It will happen for infinity, since there is no giving up.